### PR TITLE
Intl.Segmenter: Add some tests for the constructor.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

File: test/intl402/Segmenter/constructor/constructor/subclassing.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+assert.compareArray(segments(custom_segmenter.segment(value)),
+                    segments(real_segmenter.segment(value)),
+                    "Direct call");
+
+assert.compareArray(segments(Intl.Segmenter.prototype.segment.call(custom_segmenter, value)),
+                    segments(Intl.Segmenter.prototype.segment.call(real_segmenter, value)),
+                    "Indirect call");
+
+assert.sameValue(Object.getPrototypeOf(custom_segmenter), CustomSegmenter.prototype, "Prototype");
```
Comment (rwaldron) [inline test/intl402/Segmenter/constructor/constructor/subclassing.js]: Missing:

```js
assert.sameValue(Object.getPrototypeOf(CustomSegmenter), Intl.Segmenter, 'Object.getPrototypeOf(CustomSegmenter) returns Intl.Segmenter');
assert.sameValue(Object.getPrototypeOf(CustomSegmenter.prototype), Intl.Segmenter.prototype, 'Object.getPrototypeOf(CustomSegmenter.prototype) returns Intl.Segmenter.prototype');
assert(
  custom_segmenter instanceof Intl.Segmenter,
  'The result of `custom_segmenter instanceof Intl.Segmenter` is true'
);

assert.throws(TypeError, function() {
  CustomSegmenter();
});
```

Comment (Ms2ger) [reply test/intl402/Segmenter/constructor/constructor/subclassing.js]: Added.

Comment (rwaldron) [reply test/intl402/Segmenter/constructor/constructor/subclassing.js]: Thanks!

**Comment by rwaldron** [comment]
When I land these, I'm going to move them to `test/intl402/Segmenter/*`

**Comment by Ms2ger** [comment]
Please do not do that.

**Comment by rwaldron** [comment]
> Please do not do that.

Fine, but all of the test/intl402/* directories have a structure that doesn't match the convention used throughout the rest of the suite and eventually this will need to be fixed.

**Comment by Ms2ger** [comment]
Can you file an issue to discuss that?

**Pull request merged**